### DISCUSSION TITLE
connection是否应该进行线程保护？

### DISCUSSION THREAD

**Discussion by sirius-zyj** [chenshuo/muduo#685] [Q&A]
![image](https://github.com/chenshuo/muduo/assets/62042039/bc065551-6e61-4170-a822-cfdf24a49be3)
此处一增一减，是否应该进行线程保护，比如加锁？

**Comment by chenshuo** [comment]
不用。
因为这两个函数开头都有：

```
loop_->assertInLoopThread();
```

**Comment by xb1520** [comment]
**loop_->assertInLoopThread()**确保在当前线程是loop创建时所在线程呀，一个**loop**一个线程不会产生线程安全问题的
